Sources/SwiftProtobuf/ProtobufJSONEncoding.swift

```diff
@@ -21,16 +21,32 @@ struct ProtobufJSONEncodingVisitor: ProtobufVisitor {
     private var encoder = ProtobufJSONEncoder()
     var result: String {return encoder.result}
 
-    init() {}
+    private var nameResolver: (Int) -> String?
+    private var anyTypeURL: String?
+
+    init(message: ProtobufJSONMessageBase, anyTypeURL: String? = nil) throws {
+        self.nameResolver =
+            ProtoNameResolvers.jsonFieldNameResolver(for: message)
+        self.anyTypeURL = anyTypeURL
 
-    init(message: ProtobufJSONMessageBase) throws {
         try withAbstractVisitor {(visitor: inout ProtobufVisitor) in
             try message.traverse(visitor: &visitor)
         }
     }
 
     mutating func withAbstractVisitor(clause: (inout ProtobufVisitor) throws -> ()) throws {
         encoder.startObject()
+
+        // TODO: This is a bit of a hack that exists as a workaround to make the
+        // hand-written Any serialization work with the new design. We need to
+        // generate those WKTs instead of maintaining the hand-written ones,
+        // handle the special cases differently, and then remove this.
```

Comment:
> .. because even Swift extensions can't override behaviors

Actually, they can.  A swift extension on a struct can override/replace a method defined in a protocol extension.  This is how Swift Protobuf allows end developers to override/replace the default hash and equality implementations, for instance.  (It defines hashValue and isEqual(to:) in a protocol whose default implementation just invokes the `_protoc_generated` version.)